pipelines/broad/arrays/imputation_beagle/ImputationBeagle.wdl

```diff
@@ -101,86 +99,124 @@ workflow ImputationBeagle {
           var_also_in_reference = CountVariantsInChunksBeagle.var_also_in_reference
       }
 
+      # create chunk without overlaps to get sites to impute
       call tasks.SubsetVcfToRegion {
         input:
-          vcf = multi_sample_vcf,
-          vcf_index = multi_sample_vcf_index,
+          vcf = CreateVcfIndex.vcf,
+          vcf_index = CreateVcfIndex.vcf_index,
           output_basename = "input_samples_subset_to_chunk",
           contig = referencePanelContig.contig,
           start = start,
-          end = end
+          end = end,
+          gatk_docker = gatk_docker
       }
 
       call tasks.SetIDs as SetIdsVcfToImpute {
         input:
           vcf = SubsetVcfToRegion.output_vcf,
           output_basename = "input_samples_with_variant_ids"
       }
+    }
+
+    call tasks.StoreChunksInfo as StoreContigLevelChunksInfo {
+      input:
+        chroms = chunk_contig,
+        starts = start,
+        ends = end,
+        vars_in_array = CountVariantsInChunksBeagle.var_in_original,
+        vars_in_panel = CountVariantsInChunksBeagle.var_also_in_reference,
+        valids = CheckChunksBeagle.valid,
+        basename = output_basename
+    }
+
+    # if any chunk for any chromosome fail CheckChunks, then we will not impute run any task in the next scatter,
+    # namely phasing and imputing which would be the most costly to throw away
+    Int n_failed_chunks_int = select_first([error_count_override, read_int(StoreContigLevelChunksInfo.n_failed_chunks)])
+    call tasks.ErrorWithMessageIfErrorCountNotZero as FailQCNChunks {
+      input:
+        errorCount = n_failed_chunks_int,
+        message = "contig " + referencePanelContig.contig + " had " + n_failed_chunks_int + " failing chunks"
+    }
+
+    Array[File] chunkedVcfsWithOverlapsForImputation = GenerateChunk.output_vcf
+    Array[File] chunkedVcfsWithoutOverlapsForSiteIds = SetIdsVcfToImpute.output_vcf
+    Array[File] chunkedVcfIndexesWithoutOverlapsForSiteIds = SetIdsVcfToImpute.output_vcf_index
+
+    scatter (i in range(num_chunks)) {
+      String chunk_basename_imputed = referencePanelContig.contig + "_chunk_" + i + "_imputed"
+      
+      # these are copies from the first scatter, but need to have a different variable name
+      Int start2 = (i * chunkLength) + 1
```

Comment:
can you reference the `start` variable from the other scatter here? (i honestly have no idea if thats possible)